### PR TITLE
Working api

### PR DESCRIPTION
**Pull request by BreadAndRoses95**
May need some tweaking to match the exact behavior wanted.

### CODE REVIEW

File: ui/app/containers/AgentPage/index.js [hunk 1]
```diff
@@ -193,8 +195,16 @@ export class AgentPage extends React.PureComponent { // eslint-disable-line reac
         position: 'bottom'
       });
     }
+    else if (this.props.agent.usePostFormat && this.props.postFormat.postFormatPayload === '') {
+      this.props.onChangePostFormatData("postFormatPayloadDefault", messages.defaultPostFormat);
+
+      // this.props.postFormat.postFormatPayload = messages.defaultPostFormat;
```
Comment (dcalvom) [inline ui/app/containers/AgentPage/index.js]: If this is not being used anymore you can remove it

File: ui/app/containers/AgentPage/index.js [hunk 2]
```diff
@@ -288,6 +300,16 @@ export class AgentPage extends React.PureComponent { // eslint-disable-line reac
                 onChange={this.props.onChangeAgentData.bind(null, 'useWebhook')}
                 checked={agent.useWebhook}
               />
+
+
+            </Row>
+            <Row style={{ marginTop: '15px' }}>
+              <Toggle
+                label={"Post format"}
```
Comment (dcalvom) [inline ui/app/containers/AgentPage/index.js]: use {messages.usePostFormat.defaultMessage} instead

File: ui/app/containers/AgentPage/messages.js [hunk 3]
```diff
@@ -127,6 +139,10 @@ export default defineMessages({
     id: 'boilerplate.containers.AgentPage.create_agent.use_webhook',
     defaultMessage: 'Webhook',
   },
+  usePostformat:{
+    id: 'boilerplate.containers.AgentPage.create_agent.use_postFormat',
+    defaultMessage: 'PostFormat',
```
Comment (dcalvom) [inline ui/app/containers/AgentPage/messages.js]: change to 'Post Format'

File: ui/app/containers/IntentPage/sagas.js [hunk 4]
```diff
@@ -170,18 +196,44 @@ function* deleteWebhook(payload) {
...
+
 function* putScenario(payload) {
   const { api, id } = payload;
   const scenarioData = yield select(makeSelectScenarioData());
   const oldScenarioData = yield select(makeSelectOldScenarioData());
   try {
-    if (!_.isEqual(scenarioData, oldScenarioData)){
+    if (!_.isEqual(scenarioData, oldScenarioData)) {
       const { agent, domain, intent, ...data } = scenarioData;
+      // delete data.usePostFormat;
```
Comment (dcalvom) [inline ui/app/containers/IntentPage/sagas.js]: This one could be deleted too

**Comment by BreadAndRoses95** [comment]
Ui part may not be exactly what you asked for, not a react expert yet :/

**Comment by BreadAndRoses95** [comment]
Working on the import/export problem

**Pull request merged**